src/main/java/swerve/ModuleIOReal.java

```diff
@@ -18,87 +22,84 @@ public class ModuleIOReal implements ModuleIO {
     private double driveMotorVelocitySetpoint;
 
     public ModuleIOReal(int driveMotorID, int angleMotorID, int encoderID,
                         double[] motionMagicConfigs, int number) {
-
         this.driveMotor = new TalonFX(driveMotorID);
         this.angleMotor = new TalonFX(angleMotorID);
 
         this.encoder = new DutyCycleEncoder(encoderID);
 
-        this.motionMagicConfigs = motionMagicConfigs;
-        this.number = number;
-
-        driveMotor.configFactoryDefault(Constants.CONFIG_TIMEOUT);
-        angleMotor.configFactoryDefault(Constants.CONFIG_TIMEOUT);
-
-        driveMotor.config_kP(0, SwerveConstants.DRIVE_kP, Constants.CONFIG_TIMEOUT);
-        driveMotor.config_kI(0, SwerveConstants.DRIVE_kI, Constants.CONFIG_TIMEOUT);
-        driveMotor.config_kD(0, SwerveConstants.DRIVE_kD, Constants.CONFIG_TIMEOUT);
-        driveMotor.config_kF(0, SwerveConstants.DRIVE_KF, Constants.CONFIG_TIMEOUT);
-        driveMotor.enableVoltageCompensation(true);
-        driveMotor.configVoltageCompSaturation(SwerveConstants.VOLT_COMP_SATURATION);
-        driveMotor.setNeutralMode(NeutralMode.Brake);
-        driveMotor.configSupplyCurrentLimit(SwerveConstants.SUPPLY_CURRENT_LIMIT);
-        driveMotor.configStatorCurrentLimit(SwerveConstants.STATOR_CURRENT_LIMIT);
-        driveMotor.setInverted(SwerveConstants.CLOCKWISE);
-
-        angleMotor.enableVoltageCompensation(true);
-        angleMotor.configVoltageCompSaturation(SwerveConstants.VOLT_COMP_SATURATION);
-        angleMotor.configNeutralDeadband(SwerveConstants.NEUTRAL_DEADBAND);
-        angleMotor.setNeutralMode(NeutralMode.Brake);
-        angleMotor.configSupplyCurrentLimit(SwerveConstants.SUPPLY_CURRENT_LIMIT);
-        angleMotor.configStatorCurrentLimit(SwerveConstants.STATOR_CURRENT_LIMIT);
-        angleMotor.setInverted(SwerveConstants.CLOCKWISE);
-
-        angleMotor.config_kP(0, motionMagicConfigs[0], Constants.CONFIG_TIMEOUT);
-        angleMotor.config_kI(0, motionMagicConfigs[1], Constants.CONFIG_TIMEOUT);
-        angleMotor.config_kD(0, motionMagicConfigs[2], Constants.CONFIG_TIMEOUT);
-        angleMotor.config_kF(0, motionMagicConfigs[3], Constants.CONFIG_TIMEOUT);
-        angleMotor.configMotionSCurveStrength((int) motionMagicConfigs[4], Constants.CONFIG_TIMEOUT);
-        angleMotor.configMotionCruiseVelocity(motionMagicConfigs[5], Constants.CONFIG_TIMEOUT);
-        angleMotor.configMotionAcceleration(motionMagicConfigs[6], Constants.CONFIG_TIMEOUT);
-        angleMotor.configAllowableClosedloopError(0, motionMagicConfigs[7], Constants.CONFIG_TIMEOUT);
-        angleMotor.configMaxIntegralAccumulator(0, motionMagicConfigs[8], Constants.CONFIG_TIMEOUT);
-        angleMotor.configClosedLoopPeakOutput(0, motionMagicConfigs[9], Constants.CONFIG_TIMEOUT);
+        TalonFXConfigurator driveMotorConfigurator = driveMotor.getConfigurator();
+        TalonFXConfigurator angleMotorConfigurator = angleMotor.getConfigurator();
```

Comment:
Do you have to create variables for these?

Reply:
That's what they did in the documentation but I'm pretty sure I don't have to.